### PR TITLE
Sonar plugin separation

### PR DESCRIPTION
**Pull request by woensug-choi**
This commit includes
- PR https://github.com/Field-Robotics-Lab/nps_uw_sensors_gazebo/pull/39 (branched from this branch) and PR https://github.com/Field-Robotics-Lab/nps_uw_sensors_gazebo/pull/43 (manually) assuming they will pass reviews.
- Removal of sonar plugin
   - New plugin repository: [Field-Robotics-Lab/nps_uw_multibeam_sonar](https://github.com/Field-Robotics-Lab/nps_uw_multibeam_sonar)
- Removal of launch/model files (they are moved to new [Field-Robotics-Lab/nps_uw_multibeam_sonar](https://github.com/Field-Robotics-Lab/nps_uw_multibeam_sonar) repository).
  - Blueview p900, Blueview m450, Seabat f50

How-to
1. Clone [Field-Robotics-Lab/nps_uw_multibeam_sonar](https://github.com/Field-Robotics-Lab/nps_uw_multibeam_sonar) repository at workspace
   - `git clone git@github.com:Field-Robotics-Lab/nps_uw_multibeam_sonar.git'
2. Clean the previous compile and recompile
   - `catkin_make clean && catkin_make`
3. Launch tests for each sonar models
   ```bash
   roslaunch nps_uw_multibeam_sonar sonar_tank_blueview_p900_nps_multibeam.launch
   roslaunch nps_uw_multibeam_sonar sonar_tank_blueview_m450_nps_multibeam.launch
   roslaunch nps_uw_multibeam_sonar sonar_tank_seabat_f50_nps_multibeam.launch
   ```

Post-merge to-do
- Edit Multibeam sonar wiki page
- Archive previous sonar wiki pages

### CODE REVIEW

**Comment by woensug-choi** [comment]
@M1chaelM Not much is left at the `src` folder. Please check if I have deleted anything that I should not.

**Comment by woensug-choi** [comment]
@dtdavi1 May I invite you to review this?

**Review by dtdavi1 — DISMISSED** [review]
Functionally, everything looks great.  Can (should) we move the SDF sonar models and demo launch files from this repository to nps_uw_multibeam_sonar as part of this merge since they're completely associated with the sonar capability?

**Comment by woensug-choi** [comment]
@dtdavi1 Sure! I will double check that it would not harm the Dave project's launches and move sdf's to the other repository.

File: launch/sonar_tank_blueview_p900_nps_multibeam.launch [hunk 1]
```diff
@@ -1,13 +1,13 @@
 <?xml version="1.0"?>
 <launch>
   <include file="$(find gazebo_ros)/launch/empty_world.launch">
-    <arg name="world_name" value="$(find nps_uw_sensors_gazebo)/worlds/sonar_tank_blueview_p900.world"/>
+    <arg name="world_name" value="$(find nps_uw_sensors_gazebo)/worlds/sonar_tank_blueview_p900_nps_multibeam.world"/>
```
Comment (mabelzhang) [inline launch/sonar_tank_blueview_p900_nps_multibeam.launch]: Just a comment as I'm trying to get things set up and running. The verbose flag (as instructed in the wiki) to the launch file isn't doing anything right now.
I added it locally:

```.diff
+  <arg name="verbose" default="false"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find nps_uw_sensors_gazebo)/worlds/sonar_tank_blueview_p900_nps_multibeam.world"/>
+    <arg name="verbose" value="$(arg verbose)"/>
  </include>
```

**Comment by mabelzhang** [comment]
> Can (should) we move the SDF sonar models and demo launch files

Agreed. It was a bit confusing that the launch file and models weren't in the new repo - the two repos have the same prefix so I didn't realize they were different names. The dependency should be listed in package.xml if the files are straddled as they are now.

**Comment by woensug-choi** [comment]
## Ready to merge
Excluded all sonar model and related launch/config/world files. They are moved to the other repo.
New launch commands are (changed from 'nps_uw_sensors_gazebo' to 'nps_uw_multibeam_sonar')
```bash
roslaunch nps_uw_multibeam_sonar sonar_tank_blueview_p900_nps_multibeam.launch
roslaunch nps_uw_multibeam_sonar sonar_tank_blueview_m450_nps_multibeam.launch
roslaunch nps_uw_multibeam_sonar sonar_tank_seabat_f50_nps_multibeam.launch
```

This should be good to go now.

**Pull request merged**